### PR TITLE
Add gitops validation for workload cluster

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/543#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/v1alpha1/gitopsconfig_types.go [hunk 1]
```diff
@@ -35,6 +37,10 @@ type Github struct {
 	Personal bool `json:"personal,omitempty"`
 }
 
+func (g *Github) ClusterRootPath() string {
```
Comment (mitalipaygude) [inline pkg/api/v1alpha1/gitopsconfig_types.go:40]: Why did we end up introducing this?

File: pkg/validations/createvalidations/gitops.go [hunk 2]
```diff
@@ -23,5 +23,18 @@ func ValidateGitOpsNameIsUnique(ctx context.Context, k validations.KubectlClient
 	if len(existingGitOps) > 0 {
 		return fmt.Errorf("gitOpsConfig %s already exists", spec.Spec.GitOpsRef.Name)
 	}
+
+	mgmtCluster, err := k.GetEksaCluster(ctx, cluster, cluster.Name)
+	if err != nil {
+		return err
```
Comment (mitalipaygude) [inline pkg/validations/createvalidations/gitops.go:29]: nit: do we want to have additional details here for the error message?

File: pkg/api/v1alpha1/gitopsconfig_types.go (hunk 1 above)
Comment (jiayiwang7) [reply pkg/api/v1alpha1/gitopsconfig_types.go:40]: to support the new gitops file structure. https://github.com/aws/eks-anywhere/issues/355 we move `flux-system` dir one level up

File: pkg/validations/createvalidations/gitops.go (hunk 2 above)
Comment (jiayiwang7) [reply pkg/validations/createvalidations/gitops.go:29]: it is sorta taken care of in the executable code

File: pkg/validations/createvalidations/gitops.go [hunk 3]
```diff
@@ -23,5 +23,18 @@ func ValidateGitOpsNameIsUnique(ctx context.Context, k validations.KubectlClient
...
+	}
+	mgmtGitOps, err := k.GetEksaGitOpsConfig(ctx, mgmtCluster.Spec.GitOpsRef.Name, cluster.KubeconfigFile, mgmtCluster.Namespace)
+	if err != nil {
+		return err
+	}
+	if mgmtGitOps.Spec.Flux.Github.ClusterRootPath() != spec.GitOpsConfig.Spec.Flux.Github.ClusterRootPath() {
+		return fmt.Errorf("gitOpsConfig.Spec.Flux.ClusterConfigPath: %s is invalid: expect workload cluster's GitOps clusterConfigPath to share the same parent directory as managaement cluster's", spec.GitOpsConfig.Spec.Flux.Github.ClusterConfigPath)
```
Comment (danbudris) [inline pkg/validations/createvalidations/gitops.go:36]:
```suggestion
		return fmt.Errorf("gitOpsConfig.Spec.Flux.ClusterConfigPath: %s is invalid: expected workload cluster's GitOps clusterConfigPath to share the same parent directory as management cluster's", spec.GitOpsConfig.Spec.Flux.Github.ClusterConfigPath)
```

File: pkg/validations/createvalidations/gitops.go [hunk 4]
```diff
@@ -10,9 +10,9 @@ import (
 	"github.com/aws/eks-anywhere/pkg/validations"
 )
 
-func ValidateGitOpsNameIsUnique(ctx context.Context, k validations.KubectlClient, cluster *types.Cluster, spec *cluster.Spec) error {
+func ValidateGitOps(ctx context.Context, k validations.KubectlClient, cluster *types.Cluster, spec *cluster.Spec) error {
```
Comment (danbudris) [inline pkg/validations/createvalidations/gitops.go:13]: hm, `ValidateGitOps` is a pretty broad name, and this method only validates the relationship between a Mgmt cluster and workload cluster gitops file structure. `ValidateGitOpsDirStructure`? `ValidateGitOpsRootPath`? 

Or keep `ValidateGitOps` as the top-level method and make a sub-method like `validateGitOpsRootPath` that's called here, so it's clear what we're validating

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/543#pullrequestreview-794662981" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/543#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/validations/createvalidations/gitops.go (hunk 3 above)
Comment (danbudris) [reply pkg/validations/createvalidations/gitops.go:36]: 🖊️

Comment (danbudris) [reply pkg/validations/createvalidations/gitops.go:36]: at one point my major was Journalism, just sayin'...

**Review by mdsgabriel — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/543#pullrequestreview-794662981" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/543#" title="Author self-approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/543#pullrequestreview-794671612" title="Approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/validations/createvalidations/gitops.go (hunk 3 above)
Comment (jiayiwang7) [reply pkg/validations/createvalidations/gitops.go:36]: lol

File: pkg/validations/createvalidations/gitops.go (hunk 4 above)
Comment (jiayiwang7) [reply pkg/validations/createvalidations/gitops.go:13]: updated!

**Comment by abhay-krishna** [comment]
New changes are detected. LGTM label has been removed.

**Pull request merged**